Modules/TPC/src/TH1ReductorTPC.cxx

```diff
@@ -0,0 +1,85 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   TH1Reductor.cxx
+/// \author Marcel Lesch
+/// \author Cindy Mordasini
+/// \author Based on the work from Piotr Konopka
+///
+
+#include <TH1.h>
+#include <vector>
+#include <TCanvas.h>
+#include <TList.h>
+#include <TAxis.h>
+#include "TPC/TH1ReductorTPC.h"
+
+namespace o2::quality_control_modules::tpc
+{
+
+void* TH1ReductorTPC::getBranchAddress()
+{
+  return &mStats;
+}
+
+const char* TH1ReductorTPC::getBranchLeafList()
+{
+  return Form("mean[%i]/D:stddev[%i]:entries[%i]", NMAXSLICES, NMAXSLICES, NMAXSLICES);
+}
+
+void TH1ReductorTPC::update(TObject* obj, std::vector<std::vector<float>>& axis, bool isCanvas)
+{
+  TList* padList = nullptr;                // List of TPads for input TCanvas.
+  int nPads = 1;                           // Number of pads in obj (Set to 1 for TH1).
+  TH1* histo = nullptr;                    // General pointer to the histogram to trend. Common for input canvas and slicer.
+  int innerAxisSize = (int)axis[0].size(); // Dimension of the inner axisDivision. Set to 1 if no slicing or if input canvas.
+
+  // Get the list of pads/histograms if an input canvas is passed.
+  if (isCanvas) {
+    auto canvas = dynamic_cast<TCanvas*>(obj);
+    padList = dynamic_cast<TList*>(canvas->GetListOfPrimitives());
```

Comment:
```suggestion
    padList = static_cast<TList*>(canvas->GetListOfPrimitives());
```
No need for `dynamic_cast`, since you know it returns a TList.

Reply:
Done as well.